project-minimization-core/src/test/kotlin/org/plan/research/minimization/core/algorithm/dd/impl/ProbabilisticDDTest.kt

```diff
@@ -1,7 +1,94 @@
 package org.plan.research.minimization.core.algorithm.dd.impl
 
+import kotlinx.coroutines.runBlocking
+import net.jqwik.api.*
+import net.jqwik.api.statistics.NumberRangeHistogram
+import net.jqwik.api.statistics.Statistics
+import net.jqwik.api.statistics.StatisticsReport
 import org.plan.research.minimization.core.algorithm.dd.DDAlgorithm
+import org.plan.research.minimization.core.algorithm.dd.DDItemInfo
+import org.plan.research.minimization.core.model.DDItem
+import org.plan.research.minimization.core.model.EmptyMonad
+import org.plan.research.minimization.core.model.Monad
+import org.plan.research.minimization.core.model.PropertyTester
+import kotlin.random.Random
+import kotlin.test.assertContentEquals
 
 class ProbabilisticDDTest : DDAlgorithmTestBase() {
     override fun createAlgorithm(): DDAlgorithm = ProbabilisticDD()
+
+    class CountingTester<M : Monad, T : DDItem>(
+        private val internal: PropertyTester<M, T>,
+    ) : PropertyTester<M, T> {
+        private var count = 0
+
+        fun getCount() = count
+
+        fun resetCount() {
+            count = 0
+        }
+
+        context(M)
+        override suspend fun test(retainedItems: List<T>, deletedItems: List<T>) =
+            internal.test(retainedItems, deletedItems).also { count++ }
+    }
+
+    data class TestCase(
+        val items: List<SomeDDItem>,
+        val target: List<SomeDDItem>,
+    )
+
+    @Provide
+    fun testCase(): Arbitrary<TestCase> {
+        val itemsA = Arbitraries.integers().between(1, 1000).map { n -> List(n) { SomeDDItem(it) } }
+        return itemsA.flatMap { items ->
+            Arbitraries.integers().between(1, items.size).map {
+                val target = items.take(it)
+                TestCase(items, target)
+            }
+        }
+    }
+
+    @Property
+    @StatisticsReport(format = NumberRangeHistogram::class)
+    fun infoHelps(@ForAll("testCase") case: TestCase, @ForAll seed: Long) {
```

Comment:
I would to also check here if the first test attempt is "good"

Reply:
Did not understand what do you mean by "good"? You mean first call to the property tester passes all important items as retained?